lib/cn/cn_tree.c

```diff
@@ -458,9 +457,6 @@ tn_samp_update_incr(struct cn_tree_node *tn, struct kvset *kvset, bool force)
     if (tn->tn_update_incr_dgen < dgen)
         tn->tn_update_incr_dgen = dgen;
 
-    if (tn->tn_biggest_kvset < cn_ns_keys(&tn->tn_ns))
-        tn->tn_biggest_kvset = cn_ns_keys(&tn->tn_ns);
-
     return true;
```

Comment:
Nabeel found where this bug was introduced (about 3 1/2 years ago!!!)
If we really want to manage these bounds for uniq keys I think we need to account for tombstone reducing the biggest kvset.  Thoughts? Do we really distrust hlog so much?  What the worse that can happen, and inopportune garbage collection that doesn't isn't very effective?


https://bitbucket.micron.com/bbdc/projects/SBUSW/repos/nf/commits/659c8f8765cc329e2030126fe9e30aa02de1b5f5#components/kvdb/source/cn/cn_tree.c